Crash when compiler emits a warning due to analyzing a dependency

**Version(s)**
1.5.4

**Describe the bug**
A crash occurs if we try to compile a file with `-Ysafe-init-global`, and the file has a dependency whose code results in a warning.

**To Reproduce**
Create a dummy sbt project, for example:
```
sbt new scala/scala3.g8
A template to demonstrate a minimal Scala 3 application

name [Scala 3 Project Template]: hello-world

Template applied in ~/hello-world
```
Write some code in `Main.scala` that causes a warning, for example:
```scala
package hello

object Main {
  val x: Int = y
  val y: Int = x
}
```
`sbt publishLocal` the project, and use it as a dependency:
```scala
//> using scala 3.6.1
//> using option -Ysafe-init-global
//> using dep "hello-world::hello-world::0.1.0-SNAPSHOT"

import hello.Main

object Test {
  val a = Main.x
}
```

`scala-cli compile`ing this file results in a crash:
```
[warn] /[redacted]/Main.scala:1:1
[warn] Access uninitialized field value y. Calling trace:
[warn] ├──      [ Main.scala:9 ]
[warn] │   ^
[warn] └──      [ Main.scala:10 ]
[warn]                    ^
Dec 16, 2024 2:58:52 AM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint handleNotification
WARNING: Notification threw an exception: {
  "jsonrpc": "2.0",
  "method": "build/publishDiagnostics",
  "params": {
    "textDocument": {
      "uri": "file:///[redacted]/Main.scala"
    },
    "buildTarget": {
      "uri": "file:/[redacted]/?id\u003dscala3_1deee6db16-9ad788c858"
    },
    "diagnostics": [
      {
        "range": {
          "start": {
            "line": 0,
            "character": 0
          },
          "end": {
            "line": 0,
            "character": 0
          }
        },
        "severity": 2,
        "message": "Access uninitialized field \u001b[33mvalue\u001b[0m \u001b[35my\u001b[0m. Calling trace:\n├── \t[ Main.scala:9 ]\n│   ^\n└── \t[ Main.scala:10 ]\n                   ^\n",
        "data": {
          "actions": []
        }
      }
    ],
    "reset": true
  }
}
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
        at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:67)
        at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.notify(GenericEndpoint.java:152)
        at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleNotification(RemoteEndpoint.java:220)
        at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:187)
        at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
        at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
        at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
        at java.base@17.0.6/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
        at java.base@17.0.6/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base@17.0.6/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base@17.0.6/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base@17.0.6/java.lang.Thread.run(Thread.java:833)
        at com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:775)
        at com.oracle.svm.core.posix.thread.PosixPlatformThreads.pthreadStartRoutine(PosixPlatformThreads.java:203)
Caused by: java.lang.reflect.InvocationTargetException
        at java.base@17.0.6/java.lang.reflect.Method.invoke(Method.java:568)
        at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
        ... 13 more
Caused by: java.nio.file.NoSuchFileException: /[redacted]/Main.scala
        at java.base@17.0.6/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:218)
        at java.base@17.0.6/java.nio.file.Files.newByteChannel(Files.java:380)
        at java.base@17.0.6/java.nio.file.Files.newByteChannel(Files.java:432)
        at java.base@17.0.6/java.nio.file.spi.FileSystemProvider.newInputStream(FileSystemProvider.java:422)
        at java.base@17.0.6/java.nio.file.Files.newInputStream(Files.java:160)
        at os.Path.getInputStream(Path.scala:594)
        at os.read$lines$stream$$anon$3.generate(ReadWriteOps.scala:363)
        at geny.Generator.foreach(Generator.scala:51)
        at geny.Generator.foreach$(Generator.scala:33)
        at os.read$lines$stream$$anon$3.foreach(ReadWriteOps.scala:361)
        at geny.Generator.toBuffer(Generator.scala:132)
        at geny.Generator.toBuffer$(Generator.scala:33)
        at os.read$lines$stream$$anon$3.toBuffer(ReadWriteOps.scala:361)
        at geny.Generator.toArray(Generator.scala:136)
        at geny.Generator.toArray$(Generator.scala:33)
        at os.read$lines$stream$$anon$3.toArray(ReadWriteOps.scala:361)
        at os.read$lines$.apply(ReadWriteOps.scala:348)
        at scala.build.ConsoleBloopBuildClient$.$anonfun$4$$anonfun$1(ConsoleBloopBuildClient.scala:188)
        at scala.Option.map(Option.scala:242)
        at scala.build.ConsoleBloopBuildClient$.$anonfun$4(ConsoleBloopBuildClient.scala:187)
        at scala.Option.flatMap(Option.scala:283)
        at scala.build.ConsoleBloopBuildClient$.printFileDiagnostic(ConsoleBloopBuildClient.scala:186)
        at scala.build.ConsoleBloopBuildClient.onBuildPublishDiagnostics$$anonfun$2(ConsoleBloopBuildClient.scala:86)
        at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
        at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
        at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:619)
        at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:617)
        at scala.collection.AbstractIterable.foreach(Iterable.scala:935)
        at scala.build.ConsoleBloopBuildClient.onBuildPublishDiagnostics(ConsoleBloopBuildClient.scala:72)
        ... 15 more
```

**Expected behaviour**
The warning is emitted without crashing. For example,
```scala
[warn] Access uninitialized field value y. Calling trace:
[warn] ├──      [ Main.scala:9 ]
[warn] │   ^
[warn] └──      [ Main.scala:10 ]
[warn]                    ^
Compiled project (Scala 3.6.1, JVM (17))
```

--- a/modules/build/src/main/scala/scala/build/ConsoleBloopBuildClient.scala
+++ b/modules/build/src/main/scala/scala/build/ConsoleBloopBuildClient.scala
@@ -4,7 +4,7 @@ import ch.epfl.scala.bsp4j
 
 import java.io.File
 import java.net.URI
-import java.nio.file.Paths
+import java.nio.file.{NoSuchFileException, Paths}
 
 import scala.build.errors.Severity
 import scala.build.internal.WrapperParams
@@ -185,12 +185,19 @@ object ConsoleBloopBuildClient {
       for {
         line <- lineOpt
         p    <- path.toOption
-        lines = os.read.lines(p)
-        line <- if (line < lines.length) Some(lines(line)) else None
+        lines =
+          try
+            os.read.lines(p)
+          catch
+            case e: NoSuchFileException =>
+              logger.message(s"File not found: $p")
+              logger.error(e.getMessage)
+              Nil
+        line <- lines.lift(line)
       } yield line
     }
     for (code <- codeOpt)
-      code.linesIterator.map(prefix + _).foreach(logger.error(_))
+      code.linesIterator.map(prefix + _).foreach(logger.error)
     val canPrintUnderline = diag.getRange.getStart.getLine == diag.getRange.getEnd.getLine &&
       diag.getRange.getStart.getCharacter != null &&
       diag.getRange.getEnd.getCharacter != null &&

--- a/modules/integration/src/test/scala/scala/cli/integration/RunScalacCompatTestDefinitions.scala
+++ b/modules/integration/src/test/scala/scala/cli/integration/RunScalacCompatTestDefinitions.scala
@@ -606,4 +606,40 @@ trait RunScalacCompatTestDefinitions {
         expect(r.out.trim() == macroSettings.mkString(", "))
       }
     }
+
+  if (actualScalaVersion.startsWith("3"))
+    test("-Ysafe-init-global doesnt crash the CLI when a dependency produces a warning") {
+      val (org, name, version) = ("hello-world", "test-safe-init-global-works", "0.0.1-SNAPSHOT")
+      val (depDir, mainDir)    = ("dep", "main")
+      val (depPackage, depClass, depMethod) = ("hello", "DepMain", "x")
+      val expectedMessage                   = "Hello, world!"
+      TestInputs(
+        os.rel / depDir / "Main.scala" ->
+          s"""//> using publish.organization $org
+             |//> using publish.name $name
+             |//> using publish.version $version
+             |package $depPackage
+             |
+             |object $depClass {
+             |  val $depMethod: Int = y
+             |  val y: Int = $depMethod
+             |}
+             |""".stripMargin,
+        os.rel / mainDir / "Main.scala" ->
+          s"""//> using option -Ysafe-init-global
+             |//> using dep $org::$name:$version
+             |
+             |import $depPackage.$depClass
+             |
+             |object Main extends App {
+             |  val a = $depClass.$depMethod
+             |  println("$expectedMessage")
+             |}
+             |""".stripMargin
+      ).fromRoot { root =>
+        os.proc(TestUtil.cli, "--power", "publish", "local", depDir, extraOptions).call(cwd = root)
+        val res = os.proc(TestUtil.cli, "run", mainDir, extraOptions).call(cwd = root)
+        expect(res.out.trim() == expectedMessage)
+      }
+    }
 }

